### PR TITLE
feat: add APIs to easily test the Pino logger

### PR DESCRIPTION
**Pull request by ruddenchaux**

### CODE REVIEW

File: package-lock.json
Comment (jsumners) [inline package-lock.json:1]: Please remove this file.

File: package.json [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+  },
+  "devDependencies": {
+    "@types/node": "^20.11.25",
+    "eslint": "^8.57.0",
+    "eslint-config-standard": "^17.1.0",
+    "eslint-plugin-import": "^2.29.1",
+    "eslint-plugin-n": "^16.6.2",
+    "eslint-plugin-node": "^11.1.0",
+    "eslint-plugin-promise": "^6.1.1",
+    "jest": "^29.7.0",
```
Comment (jsumners) [inline package.json]: We use either `tap` or `node:test`. We do not use `jest`.

File: package.json [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+    "pino": "^8.19.0",
+    "pre-commit": "^1.2.2",
```
Comment (jsumners) [inline package.json]: This should be `@fastify/pre-commit`.

File: README.md [hunk 3]
```diff
@@ -1 +1,241 @@
-# pino-test
\ No newline at end of file
+# pino-test
+
+A set of API to easily test the Pino logger.
```
Comment (jsumners) [inline README.md]:
```suggestion
Provides a set of utilities for verifying logs generated by the Pino logger.
```

File: README.md [hunk 4]
```diff
@@ -1 +1,241 @@
...
+  logger.info('hello world')
+  logger.info('hi world')
+
+  const expected = [
+    { msg: 'hello world', level: 30 },
+    { msg: 'hi world', level: 30 }
+  ]
+  await pinoTest.consecutive(stream, expected, is)
+})
+```
```
Comment (jsumners) [inline README.md]: This whole block is redundant. It doesn't seem to show anything unique about using `jest` over `tap` or `node:test`.

File: README.md [hunk 5]
```diff
@@ -1 +1,241 @@
...
+test('pino should log a info message using a own assert function', async ({ same }) => {
+  const stream = pinoTest.sink()
+  const logger = pino(stream)
+
+  logger.info('hello world')
+
+  const expected = { msg: 'hello world', level: 30 }
+  await pinoTest.once(stream, expected, same)
+})
+```
```
Comment (jsumners) [inline README.md]: The only difference between this example and the `node:test` example is the including of a second test that utilizes a custom assertion function. I think we can move that example into the `node:test` block and completely remove the `tap` example as it is redundant.

**Review by jsumners — CHANGES_REQUESTED** [review]

**Review by jsumners — APPROVED** [review]

**Review by mcollina — APPROVED** [review]
lgtm

**Comment by mcollina** [comment]
Thank you very much!

**Comment by ruddenchaux** [comment]
It is a pleasure, thank you for the code reviews!

**Pull request merged**